src/python/pants/java/nailgun_protocol.py

```diff
@@ -245,7 +245,7 @@ def isatty_to_env(cls, stdin, stdout, stderr):
     def gen_env_vars():
       for fd_id, fd in zip(STDIO_DESCRIPTORS, (stdin, stdout, stderr)):
         is_atty = fd.isatty()
-        yield (cls.TTY_ENV_TMPL.format(fd_id), bytes(int(is_atty)))
+        yield (cls.TTY_ENV_TMPL.format(fd_id), bytes([int(is_atty)]))
```

Comment:
I think this should be: `bytes(str(int(True)), 'ascii')` instead.. because what we want here is to cast a `bool` value to e.g. `b'0'`/`b'1'` to set as an env var:

```
>>> bytes([int(True)])
b'\x01'
>>> bytes(str(int(True)), 'ascii')
b'1'
```

Reply:
Should we be using the python 2 `str()` here? It seems so -- when I run `from builtins import str` I get `TypeError: unicode string argument without an encoding`.

Reply:
looks like the second arg to `bytes` might be py3 specific.. so for now this is probably the best:

```
import six
...
six.binary_type(str(int(is_atty)))
```

Reply:
(using `bytes(str(int(is_atty)))` fixes the crash)

Reply:
`bytes(str(int(is_atty)))` works too.

Reply:
Ok, thanks!

Reply:
72b0f7e uses `six.binary_type` -- thanks.